src/Symfony/Component/Messenger/Command/FailedMessagesShowCommand.php

```diff
@@ -67,23 +69,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
             throw new RuntimeException(sprintf('The "%s" receiver does not support listing or showing specific messages.', $this->getReceiverName()));
         }
 
-        if (null === $id = $input->getArgument('id')) {
-            $this->listMessages($io, $input->getOption('max'));
+        if ($input->getOption('stats')) {
+            $this->listMessagesPerClass($io, $input->getOption('max'));
+        } elseif (null === $id = $input->getArgument('id')) {
+            $this->listMessages($io, $input->getOption('max'), $input->getOption('class-filter'));
         } else {
             $this->showMessage($id, $io);
         }
 
         return 0;
     }
 
-    private function listMessages(SymfonyStyle $io, int $max)
+    private function listMessages(SymfonyStyle $io, int $max, ?string $classFilter)
     {
         /** @var ListableReceiverInterface $receiver */
         $receiver = $this->getReceiver();
         $envelopes = $receiver->all($max);
 
         $rows = [];
         foreach ($envelopes as $envelope) {
+            $currentClassName = \get_class($envelope->getMessage());
+
+            if ($classFilter && $classFilter !== $currentClassName) {
+                continue;
+            }
```

Comment:
that's break pagination and the `max` options.
I think we should at least show how many messages have been ignored